KarhooUISDK/Classes/CancelRideBehaviour.swift

```diff
@@ -12,51 +12,88 @@ import KarhooSDK
                 delegate: CancelRideDelegate? = nil,
                 alertHandler: AlertHandlerProtocol,
                 phoneNumberCaller: PhoneNumberCallerProtocol = PhoneNumberCaller()) {
         self.trip = trip
+        self.tripService = tripService
         self.delegate = delegate
         self.alertHandler = alertHandler
         self.phoneNumberCaller = phoneNumberCaller
     }
-
-    public func triggerCancelRide() {
-        showConfirmCancelRideAlert()
+    
+    public func cancelPressed() {
+        delegate?.showLoadingOverlay()
+        
+        tripService.cancellationFee(identifier: getTripIdentifier())
+            .execute(callback: { [weak self] result in
+                self?.delegate?.hideLoadingOverlay()
+                guard let self = self else {
+                    return
+                }
+                
+                if(result.isSuccess()) {
+                    self.showCancellationFeeAlert(cancellationFee: result.successValue() ?? CancellationFee())
+                } else {
+                    self.showCancellationFailedAlert()
+                }
+            })
+    }
+    
+    private func getTripIdentifier() -> String {
+        if Karhoo.configuration.authenticationMethod().isGuest() {
+            return trip.followCode
+        } else {
+            return trip.tripId
+        }
     }
 
     private func cancelBookingConfirmed() {
         delegate?.showLoadingOverlay()
 
-        delegate?.sendCancelRideNetworkRequest { [weak self] result in
-            self?.delegate?.hideLoadingOverlay()
+        let tripCancellation = TripCancellation(tripId: getTripIdentifier(), cancelReason: .notNeededAnymore)
 
-            if result.errorValue() != nil {
-                self?.showCancellationFailedAlert()
-            }
-        }
+        tripService.cancel(tripCancellation: tripCancellation)
+            .execute(callback: { [weak self] result in
+                self?.delegate?.hideLoadingOverlay()
+                guard let self = self else {
+                    return
+                }
+
+                if result.isSuccess() {
+                    _ = self.alertHandler.show(title: UITexts.Bookings.cancellationSuccessAlertTitle,
+                                          message: UITexts.Bookings.cancellationSuccessAlertMessage,
+                                          actions: [AlertAction(title: UITexts.Generic.ok, style: .default, handler: { [weak self] _ in
+                                            self?.delegate?.handleSuccessfulCancellation()
+                                          })])
+                } else {
+                    self.showCancellationFailedAlert()
+                }
+            })
     }
 
     private func callFleetPressed() {
         phoneNumberCaller.call(number: trip.fleetInfo.phoneNumber)
     }
 
-    private func showConfirmCancelRideAlert() {
+    private func showConfirmCancelRideAlert(callback: @escaping CallbackClosure<KarhooVoid>) {
```

Comment:
Do we need the callback?

Reply:
In fact, the whole method isn't required so deleted.